tests/sqlparser_snowflake.rs

```diff
@@ -0,0 +1,152 @@
+use sqlparser::ast::*;
+use sqlparser::dialect::SnowflakeDialect;
+use sqlparser::parser::ParserError;
+use sqlparser::test_utils::*;
+
+fn table_alias(alias: &str) -> TableAlias {
+    TableAlias {
+        name: Ident {
+            value: alias.to_owned(),
+            quote_style: None,
+        },
+        columns: Vec::new(),
+    }
+}
+
+fn table(name: impl Into<String>, alias: Option<TableAlias>) -> TableFactor {
+    TableFactor::Table {
+        name: ObjectName(vec![Ident::new(name.into())]),
+        alias,
+        args: vec![],
+        with_hints: vec![],
+    }
+}
+
+fn join(relation: TableFactor) -> Join {
+    Join {
+        relation,
+        join_operator: JoinOperator::Inner(JoinConstraint::Natural),
+    }
+}
+
+macro_rules! nest {
+    ($base:expr $(, $join:expr)*) => {
+        TableFactor::NestedJoin(Box::new(TableWithJoins {
+            relation: $base,
+            joins: vec![$(join($join)),*]
+        }))
+    };
+}
+
+fn sf() -> TestedDialects {
+    TestedDialects {
+        dialects: vec![Box::new(SnowflakeDialect {})],
+    }
+}
+
+fn get_from_section_from_select_query(query: &str) -> Vec<TableWithJoins> {
```

Comment:
I haven't been able to carefully review the rest, because the move of existing tests and the addition of new ones are combined in a single commit (see the request above).